modules/nextflow/src/main/groovy/nextflow/trace/WebLogObserver.groovy

```diff
@@ -188,40 +198,63 @@ class WebLogObserver implements TraceObserver{
         message.runStatus = event // deprecated to be removed
         message.utcTime = time
 
-        // Append the trace object if present
-        if (trace)
-            message.trace = trace.store
+        if (payload instanceof TraceRecord)
+                message.trace = (payload as TraceRecord).store
+        else if (payload instanceof Map)
+                message.metadata = payload as Map
+        else if (payload != null)
+            throw new IllegalArgumentException("Only TraceRecord and Manifest class types are supported: [${payload.getClass().getName()}] $payload")
+
+        /*
+        Make sure that:
+            1) Path objects are converted to String (or it throws a StackOverflow error)
+            2) Dates follow ISO 8601
+            3) Durations are in millisecond
+         */
+        def generator = new JsonGenerator.Options()
+                .addConverter(Path) { Path p, String key ->
+                        p.toString()
+                    }
+                .addConverter(Duration) { Duration d, String key ->
+                        d.durationInMillis
+                    }
+                .dateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
+                .timezone("UTC")
+                .build()
 
         // The actual HTTP request
-        httpClient.sendHttpMessage(JsonOutput.toJson(message))
+        httpClient.sendHttpMessage(generator.toJson(message))
         logHttpResponse()
     }
 
     /**
      * Asynchronous HTTP POST request wrapper.
      * @param event The workflow run status
-     * @param trace A TraceRecord object with workflow information
-     * @return A Java string, that contains the HTTP request response
+     * @param payload An additional object to send. Must be of type TraceRecord or Manifest
      */
-    protected void asyncHttpMessage(String event, TraceRecord trace = null){
-        webLogAgent.send{sendHttpMessage(event, trace)}
+    protected void asyncHttpMessage(String event, Object payload = null){
+        webLogAgent.send{sendHttpMessage(event, payload)}
     }
 
     /**
      * Little helper function that can be called for logging upon an incoming HTTP response
      */
     protected void logHttpResponse(){
         def statusCode = httpClient.getResponseCode()
-        if (statusCode == 200)
-            log.debug "Successfully send message to ${httpClient.getUrl()} -- received status code 200"
-        else {
-            def msg = """\
+        switch (statusCode) {
+            case 200..<300:
+                log.debug "Successfully send message to ${httpClient.getUrl()} -- received status code ${statusCode}."
+                break
+            case 400..<600:
+                def msg = """\
```

Comment:
Can you elaborate a bit on this error range?

Reply:
sure!

Reply:
distinguishing between client (4xx) and server error (5xx) should be sufficient, doesn't it?

Reply:
I understand the reason for 200..300 range, but then all remaining cases should e managed the same with a default (why 400 errorcode should be logger better than default one?)

Also, not 100% sure, but I think `200..<300` is expanded to a range that is a list, this for each invocation, .. just useless GC cycle. If you be preferable a `if` with a single > && < condition.

Reply:
kk